### PR TITLE
Fix/init mint amount ref

### PR DESCRIPTION
**Pull request by magiodev**
## 1. Overview

This pull request introduces a series of updates and refinements, aiming to enhance the efficiency, readability, and reliability of our testing and initialization processes. Key changes include the optimization of test syntax, standardization of initialization methods, and improvements in error handling and module functionality.

Major highlights include:

- Streamlining test procedures through advanced iteration methods, significantly reducing the need for repetitive instructions and making our tests more concise and maintainable.
- Adopting `default_init()` as the standard initialization method for most modules, with detailed comments provided where exceptions are made. This standardization facilitates easier understanding and maintenance of the code.
- Modifying the parameters in `default_init()` to optimize initial liquidity conditions, ensuring a more balanced and effective testing environment.
- Enhancing existing test scenarios and removing redundant test files, thereby improving the focus and efficiency of our testing suite.
- Introducing additional functionality and messages in various modules, such as `contract.rs` and `tick.rs`, to extend our capabilities and streamline operations.

## 2. Implementation details

1. **Optimized Test Syntax**: Streamlined test procedures by employing iteration methods, eliminating the need for repetitive instructions.
2. **Widespread Application of default_init()**: Adopted `default_init()` as the standard initialization method across most modules, with explanatory comments added in instances where it isn't used.
3. **Modified default_init() Parameters**: Adjusted the initial ticks in `default_init()` to `-5_000_000` and `500_000`, equals to spot prices `0.5` and `1.5`. This change ensures initial liquidity is provided on a `1:1` basis, creating an optimal range situation.
4. **Consolidation of Test Scenarios**: Removed the 18 Dec variant and enhanced the standard scenario to accommodate higher value operations, streamlining test processes.
5. **Elimination of Redundant Test File**: Removed `cl_proptest.rs` as it was mislabeled and contained duplicative tests. Relevant tests were already covered in `deposit_withdraw.rs` and `range.rs` test files.
6. **Added Second In-Range Position for Swaps**: Introduced a secondary in-range position within `range.rs` tests to facilitate swap operations within the specified range, necessary to modify the range.
7. **Updated `.cargo/config`**: Modified `test-tube` command to use `--test-threads=1`, limiting the number of test threads to avoid nondeterminism.
8. **Updated `.github/workflows/build.yml`**: Modified `cargo test -- --include-ignored` command to use `--test-threads=1`, limiting the number of test threads to avoid nondeterminism.
9. **Improvements in Error Handling (`error.rs`)**: Added `TickNotFoundAfterRebuild` error for enhanced error handling.
10. **Refinements in `initialize.rs` (test_tube directory)**: Asserting an initial spot price of 1.0 on `init_test_contract()`.
11. **Updates in `proptest.rs` (test_tube directory)**: Adjusted constants and test range parameters to avoid liquidityDelta and out of tick error. Reduced number of test executions from 256 to 100.
12. **Added new query Verify and execute Build tick exponent cache**: Implemented new query to sanity check the TickExpCache and one execute entrypoint to build it asserting only admin.

## 3. How to test/use

1. Compile the latest version of the codebase using `cargo build` and `cargo test-tube-build`. This ensures that all updates are correctly integrated into the build.
2. Execute `cargo test` and `cargo test-tube` commands to run the test suite. This step will help in verifying that the modifications in test syntax, initialization methods, and other updates are functioning as intended.

Make sure to report any discrepancies or issues encountered during the testing process.

## 4. Checklist

- [x] Does the Readme need to be updated?

## 6. Future Work (optional)

Looking ahead, there are several areas where our codebase can be further refined:

1. **Enhancing the Test Suite**: We aim to assert more conditions in our tests, moving away from hardcoded values to a more dynamic testing approach. This enhancement will improve test coverage and reliability.
2. **Complete Adoption of `default_init()`**: While significant progress has been made in standardizing initialization across modules, there are still areas where `default_init()` hasn't been fully adopted. Completing this integration will streamline our initialization processes.
3. **Developing Varied Test Scenarios**: Future work will involve creating a range of test scenarios, including those that deal with out-of-range conditions and other complex operations. This will ensure our system is robust and well-equipped to handle a variety of situations.

These steps are essential in our continuous effort to maintain and improve the quality and resilience of our software.

### CODE REVIEW

**Review by ajansari95 — DISMISSED** [review]

File: smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs [hunk 1]
```diff
@@ -490,20 +507,20 @@ mod tests {
         // setup the config with amount of cases, usable for setting different values on ci vs local
         #![proptest_config(ProptestConfig::with_cases(get_cases()))]
         #[test]
-        #[ignore]
+        //#[ignore]
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs:510]:
```suggestion
        #[ignore]
```

File: smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs [hunk 2]
```diff
@@ -498,12 +515,12 @@ mod tests {
             account_indexes in get_account_index_list()
         ) {
             // Creating test core
-            let (app, contract_address, cl_pool_id, admin_account) = init_test_contract(
+            let (app, contract_address, _cl_pool_id, admin_account) = init_test_contract( // TODO: evaluate using default_init() here
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs:518]:
```suggestion
            let (app, contract_address, _cl_pool_id, admin_account) = init_test_contract( 
            // TODO: evaluate using default_init() here
```

File: smart-contracts/contracts/cl-vault/src/test_tube/range.rs [hunk 3]
```diff
@@ -26,7 +26,7 @@ mod test {
     // #[test]
     // #[ignore]
     fn move_range_works() {
-        let (app, contract, cl_pool_id, admin) = init_test_contract(
+        let (app, contract, cl_pool_id, admin) = init_test_contract( // TODO: Evaluate creating a default_init() variant i.e. out_of_range_init()
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/range.rs:29]:
```suggestion
        let (app, contract, cl_pool_id, admin) = init_test_contract( 
        // TODO: Evaluate creating a default_init() variant i.e. out_of_range_init()
```

File: smart-contracts/contracts/cl-vault/src/test_tube/range.rs [hunk 4]
```diff
@@ -125,7 +149,7 @@ mod test {
     #[test]
     #[ignore]
     fn move_range_same_single_side_works() {
-        let (app, contract, cl_pool_id, admin) = init_test_contract(
+        let (app, contract, cl_pool_id, admin) = init_test_contract( // TODO: Evaluate creating a default_init() variant i.e. out_of_range_init()
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/range.rs:152]:
```suggestion
        let (app, contract, cl_pool_id, admin) = init_test_contract( 
        // TODO: Evaluate creating a default_init() variant i.e. out_of_range_init()
```

File: smart-contracts/contracts/cl-vault/src/test_tube/range.rs [hunk 5]
```diff
@@ -217,7 +265,7 @@ mod test {
     #[test]
     #[ignore]
     fn test_swap_math_poc() {
-        let (app, _contract, _cl_pool_id, _admin) = init_test_contract(
+        let (app, _contract, _cl_pool_id, _admin) = init_test_contract( // TODO: Evaluate using default_init()
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/range.rs:268]:
```suggestion
        let (app, _contract, _cl_pool_id, _admin) = init_test_contract( 
        // TODO: Evaluate using default_init()
```

File: smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs [hunk 6]
```diff
@@ -167,7 +170,6 @@ mod tests {
     ) {
         let balance = get_user_shares_balance(wasm, contract_address, account); // TODO: get user shares balance
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs:171]:
```suggestion
        let balance = get_user_shares_balance(wasm, contract_address, account); 
        // TODO: get user shares balance
```

File: smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs [hunk 7]
```diff
@@ -498,12 +515,13 @@ mod tests {
             account_indexes in get_account_index_list()
         ) {
             // Creating test core
-            let (app, contract_address, cl_pool_id, admin_account) = init_test_contract(
+            let (app, contract_address, _cl_pool_id, admin_account) = init_test_contract( 
+            // TODO: evaluate using default_init() here
                 "./test-tube-build/wasm32-unknown-unknown/release/cl_vault.wasm",
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/test_tube/proptest.rs:520]:
```suggestion
                // TODO: evaluate using default_init() here
                "./test-tube-build/wasm32-unknown-unknown/release/cl_vault.wasm",
```

**Review by ajansari95 — CHANGES_REQUESTED** [review]

**Review by ajansari95 — DISMISSED** [review]

**Review by LaurensKubat — APPROVED** [review]
lgtm, some good improvements in this pr for our overall code health too

**Review by ajansari95 — DISMISSED** [review]
lgtm

File: smart-contracts/contracts/cl-vault/src/vault/admin.rs [hunk 8]
```diff
@@ -114,6 +115,20 @@ pub fn execute_update_config(
...
+// Rebuild and verify the tick exponent cache
+pub fn execute_build_tick_exp_cache(
+    deps: DepsMut,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    nonpayable(&info).map_err(|_| ContractError::NonPayable {})?;
+    assert_admin(deps.as_ref(), &info.sender)?;
+
+    build_tick_exp_cache(deps.storage)?;
+    verify_tick_exp_cache(deps.storage)?;
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/vault/admin.rs]: remove this

File: smart-contracts/contracts/cl-vault/src/vault/admin.rs [hunk 9]
```diff
@@ -123,6 +138,14 @@ mod tests {
         Addr, CosmosMsg, Decimal, Uint128,
     };
 
+    #[test]
+    fn test_execute_build_tick_exp_cache() {
+        let mut deps = mock_dependencies();
+
+        build_tick_exp_cache(&mut deps.storage).unwrap();
+        verify_tick_exp_cache(&mut deps.storage).unwrap();
```
Comment (ajansari95) [inline smart-contracts/contracts/cl-vault/src/vault/admin.rs:145]: can do here but in a assert operation to test

**Review by ajansari95 — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
lgtm 🚢

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request closed without merging**